src/services/pmo/favorite/FavoritesSync.tsx

```diff
@@ -0,0 +1,29 @@
+import type { FC } from "react"
+import { useEffect } from "react"
+import { useNetInfo } from "@react-native-community/netinfo"
+
+import { useUser } from "../../auth"
+
+import useFavorites from "./useFavorites"
+
+const useFavoritesSync = (): void => {
+  const user = useUser()
+  const { isConnected } = useNetInfo()
+  const { syncWithServer, localFavorites } = useFavorites(user?.id)
+
+  useEffect(() => {
+    if (user && isConnected && localFavorites) {
+      syncWithServer()
+    }
+  }, [isConnected, localFavorites, syncWithServer, user])
+
+  return
```

Comment:
This also sets off a little bell in my mind… it makes me think that “favorites” should be in context (so there’s only one instance) and `localFavorites` should be provided by the context.

Reply:
The way its used, there will only ever be one. That breaks the reusability of the component but also makes it a whole lot simpler to write and use, so it's a tradeoff; neither option is ideal. With how simple it is, I'd leave it like this.